### PR TITLE
build: check the go version on every make invocation

### PR DESCRIPTION
**Pull request by petermattis**
~ make GO=~/Development/go-1.5/bin/go
go1.6 required: go version go1.5.2 darwin/amd64
make: **\* [.go-version-check] Error 1

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/4598)

### CODE REVIEW

**Comment by vivekmenezes** [comment]
upgrade to go1.6 required (https://golang.org/dl/): current go version go1.5.2 darwin/amd64

LGTM

File: Makefile [hunk 1]
```diff
@@ -225,6 +225,20 @@ clean:
 protobuf:
 	$(MAKE) -C .. -f cockroach/build/protobuf.mk
 
+# The .go-version-check target is phony so that it is rebuilt every
+# time, but we create a .go-version-check file so that the include
+# directive below only complains about .go-version-check not existing
+# the first time we are run.
+.PHONY: .go-version-check
+.go-version-check:
+	@required=go1.6; \
```
Comment (bdarnell) [inline Makefile]: If we put the expected value "go1.6" in a separate file (`.go-version`? Is there any common convention we could follow for this?), then we could depend on that file in `sql/parser/Makefile` and remove the newly `.PHONY` targets.

**Comment by bdarnell** [comment]
LGTM

File: Makefile (hunk 1 above)
Comment (petermattis) [inline Makefile]: Hmm, yeah, I can put the version in a separate file. That would also eliminate the warning when make is invoked and `.go-version-check` cannot be found. Not sure I want to rollback the changes to `sql/parser/Makefile`. Regenerating the files there is fast. There isn't much reason to try and optimize that.

**Comment by petermattis** [comment]
@vivekmenezes Added a pointer to `CONTRIBUTING.md` in the error message.

File: Makefile (hunk 1 above)
Comment (petermattis) [inline Makefile]: Yeah, putting the version in a `.go-version` file is nicer. It also allow `make GOVERS=go1.5` to allow building with an older (or newer) go version for those who know what they are doing.

**Comment by bdarnell** [comment]
LGTM

**Pull request merged**